store/inputValidation.go

```diff
@@ -102,5 +104,33 @@ func (v *validItemUnmarshaler) UnmarshalJSON(data []byte) error {
 
 	validateItemTTL(&v.item, v.config.ItemMaxTTL)
 
+	if !validDepth(v.item.Data, v.config.ItemDataMaxDepth) {
+		return errInvalidItemDataDepth
+	}
+
 	return nil
 }
+
+// validDepth returns true if the maximum depth in data is at
+// most maxDepth. False otherwise.
+func validDepth(data map[string]interface{}, maxDepth uint) bool {
+	return validDepthHelper(data, 0, maxDepth)
+}
```

Comment:
I'm a little confused about this - if we have a non-nil map, shouldn't the depth start at 1?

I can't find anything helpful that outright says this, but most code examples I find show it...
https://stackoverflow.com/a/29006786
https://stackoverflow.com/a/20938305
https://stackoverflow.com/a/16075976